### PR TITLE
Atomic file output app

### PR DESCRIPTION
**Pull request by otterc**
@tweise Please review

### CODE REVIEW

File: exactly-once/src/main/java/com/example/myapexapp/AtomicFileOutputApp.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    Application.UniqueCounterFlat count = dag.addOperator("count", new Application.UniqueCounterFlat());
+
+    FileWriter fileWriter = dag.addOperator("fileWriter", new FileWriter());
+
+    ConsoleOutputOperator cons = dag.addOperator("console", new ConsoleOutputOperator());
+    dag.addStream("words", kafkaInput.outputPort, count.data);
+    dag.addStream("counts", count.counts, fileWriter.input, cons.input);
+  }
+
+  public static class FileWriter extends AbstractFileOutputOperator<KeyValPair<String, Integer>>
```
Comment (tweise) [inline exactly-once/src/main/java/com/example/myapexapp/AtomicFileOutputApp.java:46]: Would be nice to add a comment block that covers the important aspects of this extension. Also looking at the amount of code we should think about improvements in Malhar.

**Comment by tweise** [comment]
Did you run the application on the cluster? Example of configuration in properties.xml would be good.

File: exactly-once/src/main/java/com/example/myapexapp/AtomicFileOutputApp.java (hunk 1 above)
Comment (otterc) [inline exactly-once/src/main/java/com/example/myapexapp/AtomicFileOutputApp.java:46]: The amount of code will be much less if we rotate file based on size or windows and finalization is automatic. In this example I am request to finalize when there isn't any input.

Maybe that isn't needed and I can rotate based on windows.

Comment (tweise) [inline exactly-once/src/main/java/com/example/myapexapp/AtomicFileOutputApp.java:46]: It's good as an example as long as we explain the situation.

**Comment by otterc** [comment]
I added an app test based on the existing application test. This uses Kafka input as the existing one.
In order to run it on the cluster I need to know the kafka setup on the cluster. Are kafka configurations available anywhere?

**Pull request merged**